### DISCUSSION TITLE
False positive for CVE-2023-33246 in apache/rocketmq:4.9.7

### DISCUSSION THREAD

**Discussion by aleksandradrobnjak** [aquasecurity/trivy#5352] [False Detection]
### IDs

CVE-2023-33246

### Description

For CVE-2023-33246 it is reported here:
https://lists.apache.org/thread/1s8j2c8kogthtpv3060yddk03zq0pxyp
https://nvd.nist.gov/vuln/detail/CVE-2023-33246#range-9392213
https://avd.aquasec.com/nvd/2023/cve-2023-33246/
that
> To prevent these attacks, users are recommended to upgrade to version 5.1.1 or above for using RocketMQ 5.x or **4.9.6 or above** for using RocketMQ 4.x .

Trivy v0.45.1 image scanner reports this vulnerability in apache/rocketmq:4.9.7 

### Reproduction Steps

```bash
% docker pull apache/rocketmq:4.9.7

% trivy image --severity CRITICAL apache/rocketmq:4.9.7
2023-10-09T11:56:20.101+0200	INFO	Vulnerability scanning is enabled
2023-10-09T11:56:20.101+0200	INFO	Secret scanning is enabled
2023-10-09T11:56:20.101+0200	INFO	If your scanning is slow, please try '--scanners vuln' to disable secret scanning
2023-10-09T11:56:20.101+0200	INFO	Please see also https://aquasecurity.github.io/trivy/v0.45/docs/scanner/secret/#recommendation for faster secret detection
2023-10-09T11:56:23.570+0200	INFO	JAR files found
2023-10-09T11:56:23.570+0200	INFO	Analyzing JAR files takes a while...
2023-10-09T11:56:23.863+0200	INFO	Detected OS: centos
2023-10-09T11:56:23.863+0200	INFO	Detecting RHEL/CentOS vulnerabilities...
2023-10-09T11:56:23.905+0200	INFO	Number of language-specific files: 1
2023-10-09T11:56:23.905+0200	INFO	Detecting jar vulnerabilities...

...
Java (jar)
...
Library
org.apache.rocketmq:rocketmq-broker
(rocketmq-broker-4.9.7.jar)  

org.apache.rocketmq:rocketmq-client  
(rocketmq-client-4.9.7.jar)  

Vulnerability 
CVE-2023-33246

Severity
CRITICAL

Status
fixed

Installed Version
4.9.7

Fixed Version
5.1.1 

Title  
Apache RocketMQ may have remote code execution vulnerability
when using update configuration...
https://avd.aquasec.com/nvd/cve-2023-33246
```

### Target

Container Image

### Scanner

Vulnerability

### Target OS

macOS Sonoma

### Debug Output

```bash
.
```

### Version

```bash
Version: 0.45.1
Vulnerability DB:
  Version: 2
  UpdatedAt: 2023-10-09 06:16:04.560256506 +0000 UTC
  NextUpdate: 2023-10-09 12:16:04.560256006 +0000 UTC
  DownloadedAt: 2023-10-09 09:02:15.802913 +0000 UTC
Java DB:
  Version: 1
  UpdatedAt: 2023-10-09 00:53:28.861761628 +0000 UTC
  NextUpdate: 2023-10-12 00:53:28.861761128 +0000 UTC
  DownloadedAt: 2023-10-09 09:02:56.517235 +0000 UTC
```

### Checklist

- [ ] Read [the documentation regarding wrong detection](https://aquasecurity.github.io/trivy/dev/community/contribute/discussion/#false-detection)
- [ ] Ran Trivy with `-f json` that shows data sources and confirmed that the security advisory in data sources was correct

**Comment by nikpivkin** [comment]
Hi @aleksandradrobnjak !

The source for this vulnerability is the Gitlab Advisory Database.
```bash
cat trivy.report.json| jq '.Results[].Vulnerabilities[] | select(.VulnerabilityID == "CVE-2023-33246") | {id: .VulnerabilityID, pkg: .PkgName, source: .DataSource.ID, ver: .InstalledVersion}'
{
  "id": "CVE-2023-33246",
  "pkg": "org.apache.rocketmq:rocketmq-broker",
  "source": "glad",
  "ver": "4.9.7"
}
{
  "id": "CVE-2023-33246",
  "pkg": "org.apache.rocketmq:rocketmq-client",
  "source": "glad",
  "ver": "4.9.7"

```

This vulnerability affects artifacts [maven/org.apache.rocketmq/rocketmq-broker](https://gitlab.com/gitlab-org/security-products/gemnasium-db/-/blob/master/maven/org.apache.rocketmq/rocketmq-broker/CVE-2023-33246.yml?ref_type=heads) and [maven/org.apache.rocketmq/rocketmq-client](https://gitlab.com/gitlab-org/security-products/gemnasium-db/-/blob/master/maven/org.apache.rocketmq/rocketmq-client/CVE-2023-33246.yml?ref_type=heads) of all versions below 5.1.1. Which is true, since the fix was only [released](https://github.com/apache/rocketmq/commit/c3ada731405c5990c36bf58d50b3e61965300703) for the artifact `maven/org.apache.rocketmq/rocketmq-namesrv` (for branch 4.9.x).  And according to the [source code](https://github.com/apache/rocketmq/blob/4.9.x/broker/src/main/java/org/apache/rocketmq/broker/processor/AdminBrokerProcessor.java#L526C24-L526C24), the `rocketmq-broker` and `rocketmq-client` artifacts of versions 4.9.x are still affected by the vulnerability.

**Reply by DmitriyLewen** [reply]
@nikpivkin  I checked some related links and found no information about `client` package.
Can you confirm that the `client` package is also vulnerable?

**Reply by nikpivkin** [reply]
@DmitriyLewen Yes, I didn't see any references related to the `client` module either. I created a discussion to clarify some issues related to this vulnerability. Hopefully they will respond. https://github.com/apache/rocketmq/discussions/7430
